operator/src/main/java/oracle/kubernetes/operator/WaitForReadyStep.java

```diff
@@ -160,13 +168,19 @@ private void checkUpdatedResource(Packet packet, AsyncFiber fiber, Callback call
     fiber
         .createChildFiber()
         .start(
-            createReadAndIfReadyCheckStep(callback),
+            createReadAndIfReadyCheckStep(callback, packet),
             packet.clone(),
             null);
   }
 
-  private Step createReadAndIfReadyCheckStep(Callback callback) {
-    return createReadAsyncStep(getName(), getNamespace(), getDomainUid(), resumeIfReady(callback));
+  private Step createReadAndIfReadyCheckStep(Callback callback, Packet packet) {
+    if (initialResource != null) {
+      return createReadAsyncStep(getName(), getNamespace(), getDomainUid(), resumeIfReady(callback));
+    } else {
+      DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
+      return createReadAsyncStep(resourceName, Optional.ofNullable(info).map(i -> i.getNamespace()).orElse(null),
```

Comment:
do we need to add check to ensure resourceName is not null?

Reply:
The resourceName and initialResource are final variables and are initialized in the constructor. I'm assuming that either initialResource or resourceName will be non-null for the resource that needs to be checked. I can add a non-null check to be safe but I wasn't sure if that's going to be necessary. Thanks.